src/main/java/org/truffleruby/parser/TranslatorEnvironment.java

```diff
@@ -344,7 +344,7 @@ public TranslatorEnvironment getSurroundingMethodEnvironment() {
         return methodParent;
     }
 
-    public boolean shouldWarnYieldInModuleBody() {
+    public boolean isModuleBody() {
```

Comment:
There is already a isModuleBody() method above.
Since there is only one usage of this, I suggest calling .getSurroundingMethodEnvironment().isModuleBody() in the caller (BodyTranslator.java).

Reply:
Had the same feelings, that we might simplify it

Reply:
Done